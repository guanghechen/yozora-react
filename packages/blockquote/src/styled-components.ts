import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
