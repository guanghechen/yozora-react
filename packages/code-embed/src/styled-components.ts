import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
