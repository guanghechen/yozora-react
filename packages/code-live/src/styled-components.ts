import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
