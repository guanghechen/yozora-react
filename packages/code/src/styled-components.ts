import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
