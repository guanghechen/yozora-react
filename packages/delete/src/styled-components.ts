import 'styled-components'
import type { YozoraDeleteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
