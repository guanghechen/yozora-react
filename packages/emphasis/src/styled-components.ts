import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
