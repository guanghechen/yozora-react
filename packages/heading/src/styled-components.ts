import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
