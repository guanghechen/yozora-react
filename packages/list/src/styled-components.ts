import 'styled-components'
import type { YozoraListTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    list?: YozoraListTheme
  }
}
