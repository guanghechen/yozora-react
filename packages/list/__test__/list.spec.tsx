import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import List from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a simple content', () => {
    const text = 'Hello, world!'
    for (const ordered of [false, true]) {
      const wrapper = render(
        <List ordered={ ordered } start={ 0 }>
          <li key={ 0 }>{ text }</li>
        </List>
      )
      expect(wrapper.text()).toEqual(text)
    }
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-list'
    for (const ordered of [false, true]) {
      const wrapper = render(
        <List ordered={ ordered } start={ 0 } className={ className }>
          <li key={ 0 }><span>{ text }</span></li>
        </List>
      )
      expect(wrapper.hasClass(className)).toEqual(true)
      expect(wrapper.text()).toEqual(text)
    }
  })

  it('children is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(
          <List ordered={ true } start={ 1 } data-value="waw">
            { value }
          </List >
        )
      }).toThrow(/Failed prop type: The prop `children` is marked as required/i)
    }
  })

  it('forward ref', () => {
    for (const ordered of [false, true]) {
      const ref = React.createRef<HTMLUListElement | HTMLOListElement>()
      const wrapper = mount(
        <List ordered={ ordered } ref={ ref } start={ 1 } type="a" data-value="waw">
          <li key={ 0 }>First: Good afternoon!</li>
          <li key={ 1 }>Second: Good night!</li>
        </List>
      )
      expect(wrapper.getDOMNode()).toEqual(ref.current)
      expect(wrapper.getDOMNode().getAttribute('data-value')).toEqual('waw')
    }
  })

  it('snapshot', () => {
    for (const ordered of [false, true]) {
      const wrapper = render(
        <List ordered={ ordered } start={ 3 } type="a">
          <li key={ 0 }>apple</li>
          <li key={ 1 }>banana</li>
          <li key={ 2 }>cat</li>
        </List>
      )
      expect(wrapper).toMatchSnapshot(ordered ? 'ol' : 'ul')
    }
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        list: {
          color: 'red',
          padding: '0 1rem',
          margin: 18,
          // lineHeight: 1.5,
        }
      }
    }

    for (const ordered of [false, true]) {
      const wrapper = render(
        <ThemeProvider theme={ theme }>
          <List ordered={ ordered } start={ 3 } type="a">
            <li key={ 0 }>apple</li>
            <li key={ 1 }>banana</li>
            <li key={ 2 }>cat</li>
          </List>
        </ThemeProvider>
      )
      expect(wrapper).toMatchSnapshot(ordered ? 'ol' : 'ul')
    }
  })
})
