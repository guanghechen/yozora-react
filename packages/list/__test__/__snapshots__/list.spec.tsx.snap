// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot with theme: ol 1`] = `
.c0 {
  color: red;
  padding: 0 1rem;
  margin: 18;
  line-height: 2;
}

<ol
  class="c0"
  start="3"
  type="a"
>
  <li>
    apple
  </li>
  <li>
    banana
  </li>
  <li>
    cat
  </li>
</ol>
`;

exports[`basic rendering case snapshot with theme: ul 1`] = `
.c0 {
  color: red;
  padding: 0 1rem;
  margin: 18;
  line-height: 2;
}

<ul
  class="c0"
  start="3"
  type="a"
>
  <li>
    apple
  </li>
  <li>
    banana
  </li>
  <li>
    cat
  </li>
</ul>
`;

exports[`basic rendering case snapshot: ol 1`] = `
.c0 {
  color: inherit;
  padding: 0 0 0 1.2em;
  margin: 0 0 1em;
  line-height: 2;
}

<ol
  class="c0"
  start="3"
  type="a"
>
  <li>
    apple
  </li>
  <li>
    banana
  </li>
  <li>
    cat
  </li>
</ol>
`;

exports[`basic rendering case snapshot: ul 1`] = `
.c0 {
  color: inherit;
  padding: 0 0 0 1.2em;
  margin: 0 0 1em;
  line-height: 2;
}

<ul
  class="c0"
  start="3"
  type="a"
>
  <li>
    apple
  </li>
  <li>
    banana
  </li>
  <li>
    cat
  </li>
</ul>
`;
